pkg/workflows/sdk/v2/promise.go

```diff
@@ -0,0 +1,58 @@
+package sdk
+
+import (
+	"sync"
+)
+
+type Promise[T any] interface {
+	Await() (T, error)
```

Comment:
There's discussion around the name of Await. Summarizing other potentials:

- Get()
-() (just a function, this interface goes away. Then isn't needed, we just wraps the function directly)
- Recv() (concern is that it may imply multiple results, and it's unclear what we should do in that case.